.github/workflows/test.yml

```diff
@@ -0,0 +1,23 @@
+name: Build and Test
+on: [ pull_request ]
+
+jobs:
+  build_and_test:
+    runs-on: ubuntu-latest
+    name: Build and Test
+    steps:
+    - name: Checkout this commit
+      uses: actions/checkout@v2
+      with:
+        fetch-depth: 0
+
+    - name: C++ Style Check (compiled)
+      uses: Dal-Papa/actions-uncrustify@v1.2
+
+    - run: sudo apt-get update
+    - run: sudo apt-get install software-properties-common
+    - run: sudo apt-get update
+    - run: sudo apt-get install scons g++-9 libboost-dev libboost-program-options-dev libboost-regex-dev libboost-filesystem-dev libmariadbclient-dev libmariadb3 ccache
```

Comment:
Why would you build it outside of container, if it's possible to build inside container?

Reply:
Ooh so this was me copying over the travis stuff. Apparently, that's what the travis CI was doing when we created PRs. This is a separate workflow from the container build and is just meant as a test. We could just build containers on every commit in every branch (as a way of testing that the code change is good). Think that's better?

Reply:
I updated the other things and removed the WIP. I think this is ready to merge unless you think we should do something different with this test workflow.

Reply:
Yeah, we could build the container but avoid publishing it, so as not to spew junk into Docker Hub.

Reply:
I _think_ the travis stuff is still here cause it was here when I started the branch and the PR. I need to test that out.

Other than that I need to fix the test here but I'm super behind after all that time off. I'm going to merge this so we can get things publishing and fix the test when I get a chance. Feel like there's so many PRs in the pipeline 😱